ARCHITECTURE.md

```diff
@@ -0,0 +1,340 @@
+# JHipster architecture
+
+## Yeoman basis
+
+Internally JHipster uses [yeoman](https://yeoman.io) as core.
+
+## File structure
+
+- `.blueprint` - development blueprint, used to generate and manage samples
+- `.devcontainer` - vscode's devcontainer definitions
+- `.github` - github configuration
+- `.vscode` - vscode's configuration
+- `bin` - jit executable and helper
+- `cli` - (exported) cli implementation
+- `generators/*` - (exported) generators
+  - `command.m[tj]s` - cli options, cli arguments definitions
+  - `generator.m[tj]s` - generator implementation
+  - `index.m[tj]s` - generator exports. Must re-export generator as default export and the command
+  - `internal` - non-exported supporting libs
+  - `resources` - supporting resources
+  - `jdl` - generator's jdl specifications
+  - `support` - (exported) exported supporting libs
+  - `templates` - templates folder
+- `jdl` - (exported) jdl parser implementation
+- `rfcs` - (RFCs)[CONTRIBUTING.md#rfcs]
+- `test` - package tests
+- `test-integration` - CI related stuff. Samples, scripts.
+
+## Lifecycle
+
+- [Cli entrypoint](https://github.com/jhipster/generator-jhipster/blob/main/cli/jhipster.cjs)
+- [Basic environment validation](https://github.com/jhipster/generator-jhipster/blob/main/cli/cli.mjs)
+- [Cli arguments parsing and Environment bootstrap](https://github.com/jhipster/generator-jhipster/blob/main/cli/program.mjs)
+  - Lookup for Generators and Blueprints
+  - Build cli options and arguments definition
+  - Parse options and arguments
+- Run Generator (start the Environment) passing options and arguments
+- Run every task from the highest precedence priority until there is no more pending task
+
+### Priorities
+
+#### Initializing (yeoman)
+
+Initial generator information.
+
+- say generator specific hello
+- initial environment checks
+- argument and options loading
+
+```
+  get [Generator.INITIALIZING]() {
+    return this.asInitializingTaskGroup() {
+      sayHelloTask() {
+        this.log.info('Welcome to your generator');
+      },
+      envChecks() {
+        checkNode();
+        checkDocker();
+      },
+      async loadOptions() {
+        this.parseJHipsterArguments(command.arguments);
+        this.parseJHipsterOptions(command.options);
+      },
+    }
+  }
+```
+
+#### Prompting (yeoman)
+
+Prompt for configuration.
+
+TODO
+
+#### Configuring (yeoman)
+
+Check and fix configurations.
+
+```
+  get [Generator.CONFIGURING]() {
+    return this.asConfiguringTaskGroup() {
+      checkConfig() {
+        if (this.jhipsterConfigWithDefaults.reactive && this.jhipsterConfigWithDefaults.cacheProvider !== 'no') {
+          this.log.warn("Reactive applications doesn't support cache. Disabling");
+          this.jhipsterConfig.cacheProvider = 'no';
+        }
+      },
+    }
+  }
+```
```

Comment:
I think it looks better when these code blocks are outdented and left-aligned. Also, add `js` as the language.